src/zcbor_decode.c

```diff
@@ -720,10 +726,103 @@ bool zcbor_bool_expect(zcbor_state_t *state, bool result)
 }
 
 
-bool zcbor_float32_decode(zcbor_state_t *state, float *result)
+static bool float_check(zcbor_state_t *state, uint8_t additional_val)
 {
 	INITIAL_CHECKS_WITH_TYPE(ZCBOR_MAJOR_TYPE_SIMPLE);
-	ZCBOR_ERR_IF(ADDITIONAL(*state->payload) != ZCBOR_VALUE_IS_4_BYTES, ZCBOR_ERR_FLOAT_SIZE);
+	ZCBOR_ERR_IF(ADDITIONAL(*state->payload) != additional_val, ZCBOR_ERR_FLOAT_SIZE);
+	return true;
+}
+
+
+bool zcbor_float16_bytes_decode(zcbor_state_t *state, uint16_t *result)
+{
+
+	ZCBOR_FAIL_IF(!float_check(state, ZCBOR_VALUE_IS_2_BYTES));
+
+	if (!value_extract(state, result, sizeof(*result))) {
+		ZCBOR_FAIL();
+	}
+
+	return true;
+}
+
+
+bool zcbor_float16_bytes_expect(zcbor_state_t *state, uint16_t result)
+{
+	uint16_t value;
+
+	if (!zcbor_float16_bytes_decode(state, &value)) {
+		ZCBOR_FAIL();
+	}
+	if (value != result) {
+		ERR_RESTORE(ZCBOR_ERR_WRONG_VALUE);
+	}
+	return true;
+}
+
+
+/* Float16: */
```

Comment:
nit: would it give any value to reference the RFC that defines these?

Reply:
Could be good to mention it. I added a mention of IEEE754 in the header files for the functions.